ibridges/irodsconnector/data_operations.py

```diff
@@ -185,25 +199,30 @@ def _upload_collection(session: Session, local_path: Union[str, Path],
     if not local_path.is_dir():
         raise ValueError("local_path must be a directory.")
 
-    paths = []  # (rel_path, name)
-    for root, _, files in os.walk(local_path):
-        paths.extend([(root.removeprefix(str(local_path)), f) for f in files])
-    print(paths)
-    upload_path = IrodsPath(session,
-                            str(irods_path.joinpath(local_path.name)))
-    _ = create_collection(session, upload_path)
-
-    for folder, file_name in paths:
-        # ensure iRODS collection exists
-        _ = create_collection(session, upload_path.joinpath(folder.lstrip('/')))
-        # call irods put for Path(local_path, f[0], f[1]) to destination collection
-        dest = IrodsPath(session, str(upload_path), folder.lstrip('/'))
+    source_to_dest = _create_irods_dest(local_path, irods_path)
+    for source, dest in source_to_dest:
+        _ = create_collection(session, dest.parent)
```

Comment:
One thing that we might have to worry about is absolute/relative paths. I have tested a bit with pathlib, and something like `Path(".").parent` gives the same directory as `Path(".")`. I currently can't oversee if is an issue here specifically, but might be good to keep in mind.